### PR TITLE
Add test for write with CRUD service

### PR DESCRIPTION
**Pull request by FrankEssenberger**
## Context

In this PR I added a test fro write/update and delte operation in interaction with a CAP generated service. The functionality works more or less out of the box.

However, I found that the `etag` is not returned from the server. The entity is persisted but for productive usage the CAP colleagues should add this right?

## Definition of Done

Please consider all items and remove only if not applicable.

- [x] Tests created/adjusted for your changes.
~~- [ ] Release notes updated.~~
  * Provide sufficient context so that each entry can be understood on its own.
  * Be specific about names of functions, classes, modules, etc.
  * Describe when or where this is relevant
  * Use indicative and present tense. For example, write "Provide function `name` that does X in order to Y" over "Now X can be done by calling a new function".
~~- [ ] PR title adheres to [conventional commit guidelines](https://www.conventionalcommits.org) (please note that only `fix:` and `feat:` will end up in the release notes)~~
~~- [ ] If applicable: Properly documented (JSDoc of public API)~~
~~- [ ] If applicable: Check if `node run doc` still works.~~

### CODE REVIEW

File: test-packages/e2e-tests/db/schema.cds [hunk 1]
```diff
@@ -10,7 +10,5 @@ entity TestEntity {
   DoubleProperty: Double;
   DecimalProperty: Decimal(9,2);
   DateProperty: Date;
-  TimeOfDayProperty: Time;
```
Comment (jjtang1985) [inline test-packages/e2e-tests/db/schema.cds:13]: This property is about the `Edm.TimeOfDay`, which is a valid v4 data type.

See [here](https://github.wdf.sap.corp/MA/cap-odata-v4/blob/master/edm_type_to_cap_type.md) for cap data type to edm type mapping.
See [here](http://docs.oasis-open.org/odata/odata/v4.0/csd01/part3-csdl/odata-v4.0-csd01-part3-csdl.html#_Toc355092835) for OData V4 primitive data types.

File: test-packages/e2e-tests/db/schema.cds [hunk 2]
```diff
@@ -10,7 +10,5 @@ entity TestEntity {
...
-  DataTimeOffsetDataTimeProperty: DateTime;
```
Comment (jjtang1985) [inline test-packages/e2e-tests/db/schema.cds:14]: This property is about the `Edm.DateTimeOffset`, which is a valid v4 data type.

File: packages/core/test/test-util/test-services/v4/test-service/TestComplexType.ts [hunk 3]
```diff
@@ -62,16 +62,6 @@ export interface TestComplexType {
    * @nullable
    */
   floatProperty?: number;
-  /**
-   * Time Property.
-   * @nullable
-   */
-  timeProperty?: Time;
```
Comment (jjtang1985) [inline packages/core/test/test-util/test-services/v4/test-service/TestComplexType.ts:69]: I know there is a task to remove invalid data type in v4, but maybe we can make a separate pr.

File: test-packages/e2e-tests/test/request-builder.spec.ts [hunk 4]
```diff
@@ -1,11 +1,48 @@
...
+    .getByKey(key.keyInt, key.keyString)
+    .execute(destination);
+}
+
+async function deleteEntity(key: EntityKey): Promise<void> {
+  try {
+    const fetched = await queryEntity(key);
+    return TestEntity.requestBuilder().delete(fetched).execute(destination);
+  } catch (e) {
+    // Entity was not present nothing to delete
```
Comment (jjtang1985) [inline test-packages/e2e-tests/test/request-builder.spec.ts:25]: maybe check the error message and only ignore when it's about "nothing to delete"

**Review by jjtang1985 — CHANGES_REQUESTED** [review]
Looks good.
Please consider the following:
- To make a cleaner PR, I prefer moving the vdm changes to a separate PR.
- The 2 edm types you removed are valid data type, see comments for details.
- For etag, please make a follow up ticket or create an issue in the the cap repo.

File: packages/core/test/test-util/test-services/v4/test-service/TestComplexType.ts (hunk 3 above)
Comment (FrankEssenberger) [reply packages/core/test/test-util/test-services/v4/test-service/TestComplexType.ts]: I started on a separate PR.

File: test-packages/e2e-tests/db/schema.cds (hunk 1 above)
Comment (FrankEssenberger) [reply test-packages/e2e-tests/db/schema.cds]: I will do a separate PR to remove the Edm.Time from v4.

**Review by jjtang1985 — APPROVED** [review]
